tools/federation/src/test/java/org/eclipse/rdf4j/federated/PropertyPathTests.java

```diff
@@ -85,6 +87,67 @@ public void testPropertyPath_Alternatives() throws Exception {
 		}
 	}
 
+	@Test
+	public void testPropertyPath_ExclusiveGroup() throws Exception {
+
+		prepareTest(Arrays.asList("/tests/propertypath/data1.ttl", "/tests/propertypath/data2.ttl"));
+
+		String query = "SELECT * WHERE { ?concept a skos:Concept . ?concept skos:broader+ :mammals . ?concept rdfs:label ?label}";
+
+		String actualQueryPlan = federationContext().getQueryManager().getQueryPlan(query);
+		assertQueryPlanEquals(readResourceAsString("/tests/propertypath/query_path_exclusiveGroup.qp"),
```

Comment:
I like this mechanism - simple but effective, although potentially brittle against future changes in the string representation of the query plan (it was never really designed to be a stable format, more for informing hoomans).